test/onnx/verify/negativelogliklihood_kd_dim_weighted.cpp

```diff
@@ -0,0 +1,124 @@
+                               -184.0f,
+                               -19.5f,
+                               -88.0f,
+                               -49.0f,
+                               -248.0f,
+                               -27.5f,
+                               -120.0f};
+    EXPECT(migraphx::verify::verify_rms_range(result_vector, gold));
+}
+
+TEST_CASE(negativeloglikelihoodloss_kd_sum_reduction_weighted_test)
+{
+    migraphx::program p =
+        optimize_onnx("negativeloglikelihoodloss_kd_sum_reduction_double_weighted_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape score_shape{migraphx::shape::double_type, {4, 4, 2, 2}};
+    std::vector<double> score_data(score_shape.elements());
+    std::iota(score_data.begin(), score_data.end(), 1.0);
+
+    migraphx::shape label_shape{migraphx::shape::int32_type, {4, 2, 2}};
+    std::vector<int32_t> label_data = {0, 3, 1, 2, 0, 3, 1, 2, 0, 3, 1, 2, 0, 3, 1, 2};
+    migraphx::shape weight_shape{migraphx::shape::double_type, {4}};
+    std::vector<double> weight_data = {1.0, 0.5, 2.0, 4.0};
+
+    migraphx::parameter_map pp;
+    pp["0"] = migraphx::argument(score_shape, score_data.data());
+    pp["1"] = migraphx::argument(label_shape, label_data.data());
+    pp["2"] = migraphx::argument(weight_shape, weight_data.data());
+
+    auto result = p.eval(pp).back();
+    std::vector<double> result_vector;
+    result.visit([&](auto output) { result_vector.assign(output.begin(), output.end()); });
+    std::vector<double> gold = {-1058.0};
+    EXPECT(migraphx::verify::verify_rms_range(result_vector, gold));
+}
+
+TEST_CASE(negativeloglikelihoodloss_kd_mean_reduction_weighted_test)
+{
+    using migraphx::half;
+    migraphx::program p =
+        optimize_onnx("negativeloglikelihoodloss_kd_mean_reduction_half_weighted_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape score_shape{migraphx::shape::half_type, {4, 4, 2, 2}};
+    std::vector<half> score_data(score_shape.elements());
+    std::iota(score_data.begin(), score_data.end(), half(1.0));
+
+    migraphx::shape label_shape{migraphx::shape::int32_type, {4, 2, 2}};
+    std::vector<int32_t> label_data = {0, 3, 1, 2, 0, 3, 1, 2, 0, 3, 1, 2, 0, 3, 1, 2};
+    migraphx::shape weight_shape{migraphx::shape::half_type, {4}};
+    std::vector<double> weight_data = {half(1.0), half(0.5), half(2.0), half(4.0)};
+
+    migraphx::parameter_map pp;
+    pp["0"] = migraphx::argument(score_shape, score_data.data());
+    pp["1"] = migraphx::argument(label_shape, label_data.data());
+    pp["2"] = migraphx::argument(weight_shape, weight_data.data());
+
+    auto result = p.eval(pp).back();
+    std::vector<half_float::half> result_vector;
+    result.visit([&](auto output) { result_vector.assign(output.begin(), output.end()); });
+    std::vector<half> gold = {half{-35.266666666666666}};
+    EXPECT(migraphx::verify::verify_rms_range(result_vector, gold));
+}
```

Comment:
Can you please add these 3 test cases, so we know what to expect against their published results. 
https://github.com/onnx/onnx/blob/main/docs/Operators.md#NegativeLogLikelihoodLoss
Thanks.

Reply:
Sure I can add these. I'll need to generate another onnx file for these cases

Reply:
Isn't it possible to not add new onnx files, instead just create a test case with an operator, and pass these test values, please? Thanks.

Reply:
> Isn't it possible to not add new onnx files, instead just create a test case with an operator, and pass these test values, please? Thanks.

We did need this and the case you specified actually found a bug for the previous operator this was built off of. Fix should be part of this PR now.

Reply:
Tests are also added and updated and mirror what's in the onnx spec.